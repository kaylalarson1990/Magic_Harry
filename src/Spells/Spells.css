.spellCard {
    align-items: center;
    background-color: white;
    border: .5px solid #EBECEC;
    border-radius: 3px;
    box-shadow: 2px 2px lightgrey;
    display: flex;
    flex-direction: column;
    font-family: 'Open Sans';
    height: auto;
    margin: 15px;
    width: 300px;
}

button {
    border-radius: 20px;
    cursor: pointer;
    font-size: 12px;
    margin: 25px 0;
    padding: 10px;
}

button:hover {
    background-color: #ECBA16
}

h2 {
    text-align: center;
    font-weight: 600;
}

.favorite {
    background-color: #ECBA16;
}

.favorite-btn {
    border-radius: 20px;
    cursor: pointer;
    font-size: 12px;
    margin: 25px 0;
    padding: 10px;
    width: 160px;
}
