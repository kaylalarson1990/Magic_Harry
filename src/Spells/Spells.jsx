import React, { Component } from "react";
import PropTypes from "prop-types";
import ReactCardFlip from "react-card-flip";
import "./Spells.css";

export class Spells extends Component {
  constructor() {
    super();
    this.state = {
      isFlipped: false
    };
  }

  handleClick = e => {
    e.preventDefault();
    this.setState(prevState => ({ isFlipped: !prevState.isFlipped }));
  };

  render() {
    const { spell, type, effect, id, favoriteCard, info } = this.props;
    return (
      <ReactCardFlip isFlipped={this.state.isFlipped} flipDirection="vertical">
        <article className="spellCard" key="front">
          <h2>{spell}</h2>
          <button onClick={this.handleClick}>
            Click here for more details
          </button>
          <button onClick={e => favoriteCard(info.id)} className={`favorite-btn ${info.favorite && "favorite"}`}>
              Favorite
          </button>
        </article>
        <article className="spellCard" key="back">
          <p>
            <span>Type:</span> {type}
          </p>
          <p>
            <span>Effect:</span> {effect}
          </p>
          <button onClick={this.handleClick}>Return to Front</button>
        </article>
      </ReactCardFlip>
    );
  }
}

Spells.propTypes = {
  spell: PropTypes.string,
  type: PropTypes.string,
  effect: PropTypes.string,
  id: PropTypes.number
};

export default Spells;
