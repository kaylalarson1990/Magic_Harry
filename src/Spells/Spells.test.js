import React from "react";
import { shallow } from "enzyme";
import Spells from "./Spells";
//... 23,24,25,29,30

describe("Spells", () => {
  let wrapper, mockFunc, instance;
  let event = {
    preventDefault: jest.fn()
  };

  let props = {
    spell: "Accio",
    type: "Sleeping",
    effect: "Sleeping",
    id: 1
  };

  beforeEach(() => {
    mockFunc = jest.fn();
    wrapper = shallow(<Spells {...props} />);
    instance = wrapper.instance();
  });

  it('should match snapshot', () => {
      wrapper.setState({ error: '' })
      expect(wrapper).toMatchSnapshot();
  })

  it('should be able to toggle favorite', () => {
      instance.addFavorite = mockFunc;
      wrapper.find('.favorite-btn').simulate('click');
      expect(instance.addFavorite).not.toHaveBeenCalled()
  })

  it('should set flipped to state', () => {
    expect(wrapper.state('isFlipped')).toEqual(false)
    instance.handleClick(event)
    expect(wrapper.state('isFlipped')).toEqual(true)
  })

  it('should set favorite to state', () => {
    expect(wrapper.state('favorited')).toEqual(false)
    instance.addFavorite()
    expect(wrapper.state('favorited')).toEqual(true)
  })
});
