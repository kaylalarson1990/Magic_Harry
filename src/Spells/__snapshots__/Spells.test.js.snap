// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spells should match snapshot 1`] = `
<t
  cardStyles={
    Object {
      "back": Object {},
      "front": Object {},
    }
  }
  cardZIndex="auto"
  containerStyle={Object {}}
  flipDirection="vertical"
  flipSpeedBackToFront={0.6}
  flipSpeedFrontToBack={0.6}
  infinite={false}
  isFlipped={false}
>
  <article
    className="spellCard"
    key="front"
  >
    <h2>
      Accio
    </h2>
    <button
      onClick={[Function]}
    >
      Click here for more details
    </button>
    <img
      alt="favorite icon"
      className="favorite-btn"
      onClick={[Function]}
      src="inactive.png"
    />
  </article>
  <article
    className="spellCard"
    key="back"
  >
    <p>
      <span>
        Type:
      </span>
       
      Sleeping
    </p>
    <p>
      <span>
        Effect:
      </span>
       
      Sleeping
    </p>
    <button
      onClick={[Function]}
    >
      Return to Front
    </button>
  </article>
</t>
`;
